apps/src/assetManagement/animationLibraryApi.js

```diff
@@ -90,6 +99,117 @@ export function createDefaultSpriteMetadata(listData) {
   });
 }
 
+function generateAnimationMetadataForFile(fileObject) {
+  const json = fileObject.json;
+  const png = fileObject.png;
+  return getAnimationLibraryFile(json.key)
+    .then(metadata => {
+      // Metadata contains name, frameCount, frameSize, looping, frameDelay
+      let combinedMetadata = {
+        ...metadata,
+        jsonLastModified: json.last_modified,
+        pngLastModified: png.last_modified,
+        version: png.version_id,
+        sourceUrl: `/api/v1/animation-library/level_animations/${
+          png.version_id
+        }/${metadata.name}.png`,
+        sourceSize: png.source_size
+      };
+      return Promise.resolve(combinedMetadata);
+    })
+    .catch(err => {
+      return Promise.reject(err);
+    });
+}
+
+export function buildAnimationMetadata(files) {
+  let animationMetadataByName = {};
+  let resolvedPromisesArray = [];
+  for (const [fileKey, fileObject] of Object.entries(files)) {
+    resolvedPromisesArray.push(
+      generateAnimationMetadataForFile(fileObject).then(metadata => {
+        animationMetadataByName[fileKey] = metadata;
+      })
+    );
+  }
+  return Promise.all(resolvedPromisesArray).then(() => {
+    return animationMetadataByName;
+  });
+}
+
+function buildMap(
+  animationMetadata,
+  getStandardizedContent,
+  normalizingFunction
+) {
+  let contentMap = {};
+  for (const [key, metadata] of Object.entries(animationMetadata)) {
+    let formattedArray = getStandardizedContent(metadata);
+    formattedArray.map(item => {
+      let content = normalizingFunction ? normalizingFunction(item) : item;
+      //Push name into target array, deduplicate, and sort
+      let addedSet = new Set(contentMap[item]);
+      addedSet.add(key);
+      contentMap[content] = [...addedSet].sort();
+    });
+  }
+  return contentMap;
+}
+
+function getStandardizedAliases(metadata) {
+  let aliases = [metadata.name.toLowerCase()];
+  const formattedAliases = metadata.aliases?.map(alias => {
+    alias.toLowerCase();
+  });
+  aliases = [...aliases, ...formattedAliases];
+  return aliases;
+}
+
+function getStandardizedCategories(metadata) {
+  let categories = metadata.categories;
+  // If the animation doesn't have a category, place it in a section for
+  // level-specific/hidden-from-library animations.
+  if (!categories) {
+    categories = ['level_animations'];
+  }
+  return categories;
+}
+
+// Generates the json animation manifest for the level_animations folder
+export function generateLevelAnimationsManifest() {
+  return getLevelAnimationsFiles().then(files => {
+    return buildAnimationMetadata(files).then(animationMetadata => {
+      let aliasMap = buildMap(animationMetadata, getStandardizedAliases, null);
+
+      let categoryMap = buildMap(
+        animationMetadata,
+        getStandardizedCategories,
+        category => {
+          return category.replace(' ', '_');
+        }
```

Comment:
ES6 nit: { + return aren't needed here and we can make this more readable with `category => category.replace(' ', '_')`